src/coreclr/jit/morph.cpp

```diff
@@ -14290,6 +14290,55 @@ GenTree* Compiler::fgMorphSmpOpOptional(GenTreeOp* tree)
             }
             break;
 
+#if defined(FEATURE_HW_INTRINSICS) && defined(TARGET_XARCH)
+        case GT_HWINTRINSIC:
+        {
+            GenTreeHWIntrinsic* hw = tree->AsHWIntrinsic();
+            switch (hw->gtHWIntrinsicId)
+            {
+                case NI_SSE_Xor:
+                case NI_SSE2_Xor:
+                case NI_AVX_Xor:
+                case NI_AVX2_Xor:
+                {
+                    // Optimize Xor(x, Vector_.Zero) to just x
```

Comment:
I think this needs a guard against CSE holding onto any nodes to be removed (a simple `!optValnumCSE_phase` one should suffice).

To be honest, the whole function should probably be under this guard, there's like only one switch case that is without it. But that's a separate issue.

Reply:
duh, I hate it... 🙄 
Thanks for spotting this!